src/main/java/eu/dissco/core/translator/service/BioCaseService.java

```diff
@@ -146,35 +147,43 @@ private BioCasePartResult mapToABCD(String xml, AtomicInteger processedRecords)
         }
         retrieveUnitData(xmlEventReader, processedRecords);
       }
-      if ((recordCount + recordDropped) % webClientProperties.getItemsPerRequest() != 0) {
+      if ((recordCount + recordDropped) % applicationProperties.getItemsPerRequest() != 0) {
         log.info("Received records {} does not match requested records {}. "
                 + "All records have been processed", (recordCount + recordDropped),
-            webClientProperties.getItemsPerRequest());
+            applicationProperties.getItemsPerRequest());
         return new BioCasePartResult(true, false);
       } else {
         return new BioCasePartResult(false, false);
       }
     } catch (XMLStreamException | JAXBException e) {
       log.error("Error converting response tot XML", e);
       return new BioCasePartResult(true, true);
+    } catch (ReachedMaximumLimitException e) {
+      log.warn("Reached maximum limit of {} processed specimens",
+          applicationProperties.getMaxItems());
+      return new BioCasePartResult(true, false);
     }
   }
 
   private void retrieveUnitData(XMLEventReader xmlEventReader, AtomicInteger processedRecords)
-      throws XMLStreamException, JAXBException {
+      throws XMLStreamException, JAXBException, ReachedMaximumLimitException {
     mapper.setSerializationInclusion(Include.NON_NULL);
     if (isStartElement(xmlEventReader.peek(), "DataSets")) {
       mapABCD206(xmlEventReader, processedRecords);
     }
   }
 
   private void mapABCD206(XMLEventReader xmlEventReader, AtomicInteger processedRecords)
-      throws JAXBException {
+      throws JAXBException, ReachedMaximumLimitException {
     var context = JAXBContext.newInstance(DataSets.class);
     var datasetsMarshaller = context.createUnmarshaller().unmarshal(xmlEventReader, DataSets.class);
     var datasets = datasetsMarshaller.getValue().getDataSet().get(0);
     for (var unit : datasets.getUnits().getUnit()) {
       try {
+        if (applicationProperties.getMaxItems() != null
```

Comment:
I think you need to take the chunk size into account to make sure you don't go over this limit.
if you process 1000 records at a time, and your limit is 500, then you'll overshoot before the ReachedMaximumLimitException() is thrown. 
Should chunkSize also be made into an application property?

Reply:
No that is not correct. Although I get why you would think this. The check for the maxRecords is at the individual record. This means that if we are at 500 in a batch of 1000 it will throw the exception. The exception is caught on the level of the batch so that the batch will be cut short and we will only process the 500 even if the batch is 1000. It will then finish successfully but, in case of dwca indicate how many of the total records have been processed. With Biocase we don't know the total and we will just say how many we have processed.